### ISSUE TITLE
notification incorrectly allows picking pharmacy, medication status not showing

### ISSUE THREAD

**Issue by akirakakar** [umed-jp/umed#56]
Akira has posted an issue in Discord

patientId: dde1d069-a812-42bf-9786-989616c8a43c
encounterId: 44b67b3b-4810-49a3-a0a4-949d037b8dc3

This patient has finished the appointmetn and already received medications but can still pick pharmarcy from the notification hub. Also her medication status was not showing

**Comment by fauortiz** [comment]
Details:
- Patient can pick a preferred pharmacy again, after having already gotten their medicine. 
- The pink circle indicator on the Prescription Info also does not show for the patient (they're all gray).

Notes, TODO: 
- Notification was an old one, from March 4. 
![image](https://github.com/umed-jp/umed/assets/67045972/08b61e34-6669-4433-b960-a11295bb6d51)
- Consider that maybe the patient has an older version of the app
- Consider backwards compatibility issues
- try to replicate exact scenario in dev, from what details are known

**Comment by fauortiz** [comment]
I tried to replicate the same encounter on dev. I tested the notification if would allow re-picking the pharmacy, and everything worked as intended (it correctly showed the "pharmacy already selected" alert).

details (prod):
patientId: dde1d069-a812-42bf-9786-989616c8a43c
encounterId: 44b67b3b-4810-49a3-a0a4-949d037b8dc3
appointmentId: bf2580a2-c37f-4ca8-96e6-906305526482
notification_id: b53d6840-831c-402b-b717-c13925ba32f8

I checked our encounters. some data 
	emi's notification lacks the `chosenPharmacy.appointment_end` field. mine is a timestamp.
	emi's notification lacks the `isDeleted` field. mine is `false`.
	emi's notification lacks the `isOrdered` field. mine is `null`.
	my notification lacks the `ref` field. hers is `/encounters/44b67b3b-4810-49a3-a0a4-949d037b8dc3`.

maybe it's a backwards compatibility issue related to these new fields, since the notification is from March.
